controllers/components/elasticsearch_connector.go

```diff
@@ -48,6 +52,37 @@ func (es ElasticsearchConnector) Delete() (err error) {
 }
 
 func (es *ElasticsearchConnector) CheckDeviation() (problem, err error) {
+	found, err := es.Exists()
+	if err != nil {
+		return nil, errors.Wrap(err, 0)
+	}
+	if !found {
+		return fmt.Errorf("ElasticsearchConnector named, %s, does not exist.", es.Name()), nil
+	}
+
+	esConPtr, err := es.KafkaClient.GetConnector(es.Name())
+
+	if err != nil {
+		return nil, fmt.Errorf("Error encountered when getting ElasticsearchConnector: %w", err)
+	}
+	if esConPtr != nil {
```

Comment:
I think a problem should be returned if esConPtr == nil

Reply:
Fixed and also updated the code with checks pointed out in the DebeziumConnector's PR review.